core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -702,61 +705,61 @@ class KafkaApis(val requestChannel: RequestChannel,
   /*
    * Handle an offset fetch request
    */
-
   def handleOffsetFetchRequest(request: RequestChannel.Request) {
-    val offsetFetchRequest = request.requestObj.asInstanceOf[OffsetFetchRequest]
+    val header = request.header
+    val offsetFetchRequest = request.body.asInstanceOf[OffsetFetchRequest]
 
+    val responseHeader = new ResponseHeader(header.correlationId)
+    val offsetFetchResponse =
     // reject the request immediately if not authorized to the group
```

Comment:
This comment is perhaps a bit out of date with the latest code (we do reject the request, but we no longer do the early return, so `immediately` seems a bit misleading).